### ISSUE TITLE
RFE Aimed shot window QoL

### ISSUE THREAD

**Issue by DynamicWorlds** [MegaMek/megamek#5896]
### Environment
v0.50.0-SNAPSHOT (2024-08-12 evening's nightly) both standalone and mekqh megamek
Win 10
Java Vendor: Eclipse Adoptium
Java Version: 17.0.11

### Description
Request to have the aimed shot window only come up once an enemy unit is targeted rather than every single time a new friendly unit is selected.  Needing to close it repeatedly through the firing phase is rather tedious and slows the game down considerably.

### Files

**Comment by Sleet01** [comment]
This may actually be related to #5870 ; it seems that we cache (or pre-select) a target for every unit's firing phase.

**Comment by gsparks3** [comment]
Yeah, AFAIK units default to whatever their target was on the previous turn (which includes popping up the "Hey, do you want to aim?" window if said previous target is now immobile).

**Comment by DynamicWorlds** [comment]
I believe both of you are correct in diagnosis of what's happening.  I'd still like it to not come up unless I actively target the unit that turn, if possible.  It's a lower priority than fixing bugs for 0.50, but I wanted to mention it as, for me at least, it would be a significant QoL improvement.

**Comment by Tecmes** [comment]
At some point I was told there is a reason for this, but I still can't get it, and it remains for me the #2 QoL annoying thing remaining in MM...
